plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtDomainXMLParser.java

```diff
@@ -196,6 +198,18 @@ public boolean parseDomainXML(String domXML) {
                 diskDefs.add(def);
             }
 
+            NodeList memBalloons = devices.getElementsByTagName("memballoon");
+            if ((memBalloons != null) && (memBalloons.getLength() != 0)) {
+                Element memBalloon = (Element)memBalloons.item(0);
+                String model = memBalloon.getAttribute("model");
+                MemBalloonDef def = new MemBalloonDef();
+                if (model.equalsIgnoreCase("virtio")) {
+                    String statsPeriod = getAttrValue("stats", "period", memBalloon);
+                    def.defVirtioMemBalloon(statsPeriod);
+                }
+                memBalloonDef = def;
+            }
+
```

Comment:
can this go in a separate method, please?

Reply:
Done.